core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -69,17 +69,9 @@
 
 public abstract class StreamInput extends InputStream {
 
-    private final NamedWriteableRegistry namedWriteableRegistry;
-
     private Version version = Version.CURRENT;
 
-    protected StreamInput() {
-        this.namedWriteableRegistry = new NamedWriteableRegistry();
-    }
-
-    protected StreamInput(NamedWriteableRegistry namedWriteableRegistry) {
-        this.namedWriteableRegistry = namedWriteableRegistry;
-    }
+    protected StreamInput() { }
```

Comment:
This looks like a leftover indeed an I think it can be removed, but maybe @javanna can take a second look?